services/device-registry-file/src/main/java/org/eclipse/hono/deviceregistry/file/FileBasedRegistrationService.java

```diff
@@ -520,6 +527,120 @@ private JsonObject convertDevice(final String deviceId, final Device payload) {
         return identities.computeIfAbsent(tenantId, id -> new ConcurrentHashMap<>());
     }
 
+    /**
+     * Return all devices for a tenant.
```

Comment:
IMHO it doesn't return all devices for a tenant. The JavaDoc from the parent class could be well reused by using {@inheritDoc} and then adding your additional comment regarding in-memory storage and paging inconsistency.

Reply:
Good catch !